cmd/fs/helpers.go

```diff
@@ -46,6 +48,116 @@ func filerForPath(ctx context.Context, fullPath string) (filer.Filer, string, er
 	return f, path, err
 }
 
+const dbfsPrefix string = "dbfs:/"
+const volumesPefix string = "dbfs:/Volumes"
+
 func isDbfsPath(path string) bool {
-	return strings.HasPrefix(path, "dbfs:/")
+	return strings.HasPrefix(path, dbfsPrefix)
+}
+
+func getValidArgsFunction(
+	pathArgCount int,
+	onlyDirs bool,
+	filerForPathFunc func(ctx context.Context, fullPath string) (filer.Filer, string, error),
+	mustWorkspaceClientFunc func(cmd *cobra.Command, args []string) error,
+) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		cmd.SetContext(root.SkipPrompt(cmd.Context()))
+
+		err := mustWorkspaceClientFunc(cmd, args)
+
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		filer, path, err := filerForPathFunc(cmd.Context(), toComplete)
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		isDbfsPath := isDbfsPath(toComplete)
+
+		wsc := root.WorkspaceClient(cmd.Context())
+		_, err = wsc.CurrentUser.Me(cmd.Context())
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		completer := completer.NewCompleter(cmd.Context(), filer, onlyDirs)
+
+		if len(args) >= pathArgCount {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		completions, dirPath, directive := completer.CompletePath(path)
+
+		for i := range completions {
+			completions[i] = prependDirPath(dirPath, completions[i], !isDbfsPath)
+
+			if isDbfsPath {
+				// The dirPath will start with a "/", so we'll prefix the completions with
+				// the selectedPrefix without a trailing "/"
+				prefix := dbfsPrefix[:len(dbfsPrefix)-1]
+				completions[i] = fmt.Sprintf("%s%s", prefix, completions[i])
+			} else {
+				completions[i] = handleLocalPrefix(toComplete, completions[i])
+			}
+		}
+
+		// If the path is a dbfs path, we try to add the dbfs:/Volumes prefix suggestion
+		if isDbfsPath && strings.HasPrefix(volumesPefix, toComplete) {
+			completions = append(completions, volumesPefix)
+		}
+
+		// If the path is local, we try to add the dbfs:/ prefix suggestion
+		if !isDbfsPath && strings.HasPrefix(dbfsPrefix, toComplete) {
+			completions = append(completions, dbfsPrefix)
+		}
+
+		return completions, directive
+	}
+}
+
+// Prepend directory path to completion name:
+// - Don't add a separator if the directory path is empty
+// - Don't add a separator if the dir path already ends with a separator
+// - Support \ as separator for local Windows paths
+// Note that we don't use path utilities to concatenate the path and name
+// because we want to preserve the formatting of whatever path the user has
```

Comment:
The problem is that it would break autocompletion in cases where the user hasn't entered a perfect path, like using double slashes "dbfs://" or using "/" on windows

Reply:
But I think it's fine to do so and it's in line with how `sync` command does autocompletions now if I'm not mistaken

Reply:
`sync` actually also tolerates wonky paths because it relies on the webapp for the path joining

Reply:
@andersrexdb sorry, I'm not really following here, could you link to the code where sync command does similar path interpolations?

Reply:
My bad, the `sync` command itself tolerates wonky paths, but the autocomplete chokes on them. I can go ahead and remove special handling of different syntaxes.